src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -597,7 +598,22 @@ private static FirebaseOptions getOptionsFromEnvironment() throws IOException {
       parser = jsonFactory.createJsonParser(reader);
     }
     parser.parseAndClose(builder);
-    builder.setCredentials(GoogleCredentials.getApplicationDefault());
+    builder.setDeferredCredentials(DEFAULT_OR_EMPTY_CREDS);
     return builder.build();
   }
+
+  private static final DeferredApplicationCredentials DEFAULT_OR_EMPTY_CREDS =
+      new DeferredApplicationCredentials(new CredentialsGenerator() {
+        @Override
+        public GoogleCredentials generate() {
```

Comment:
I think you could get rid of this interface and make `setDeferredCredentials` take a Supplier instead. When you do that, it would also make sense to me to rename it back to setCredentials (as an overload).